bin/reth/src/node/mod.rs

```diff
@@ -339,13 +355,38 @@ impl Command {
         .map_err(Into::into)
     }
 
+    async fn fetch_or_lookup_tip(
+        &self,
+        db: Arc<Env<WriteMap>>,
+        fetch_client: Arc<FetchClient>,
+        tip: H256,
+    ) -> Result<u64, reth_interfaces::Error> {
+        if let Some(number) = db.view(|tx| tx.get::<tables::HeaderNumbers>(tip))?? {
+            debug!(target: "reth::cli", ?tip, number, "Successfully looked up tip in the database");
+            return Ok(number)
+        }
+
+        debug!(target: "reth::cli", ?tip, "Fetching tip header from the network.");
+        loop {
+            match get_single_header(fetch_client.clone(), BlockHashOrNumber::Hash(tip)).await {
+                Ok(tip_header) => {
+                    debug!(target: "reth::cli", ?tip, number = tip_header.number, "Successfully fetched tip");
+                    return Ok(tip_header.number)
+                }
+                Err(error) => {
+                    error!(target: "reth::cli", %error, "Failed to fetch the tip. Retrying...");
+                }
+            }
+        }
```

Comment:
why do we need to fetch this in the first place?

Reply:
To be able to instantiate the pipeline with the max_block parameter if `debug.tip` was provided. otherwise, the pipeline doesn't have a terminating condition

Reply:
no, because the tip was already stored in the db. First, I attempt the db lookup, if the tip is not there, then i fetch

Reply:
i should prob change the fn name from `fetch_or_lookup_tip` to `lookup_or_fetch_tip` to reflect the order of operations